src/Microsoft.Extensions.Primitives/CompositeChangeToken.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.Extensions.Primitives
+{
+    /// <summary>
+    /// An <see cref="IChangeToken"/> which represents one or more <see cref="IChangeToken"/> instances.
+    /// </summary>
+    public class CompositeChangeToken : IChangeToken
+    {
+        private static readonly Action<object> _onChangeDelegate = OnChange;
+        private CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
```

Comment:
We could lazy initialize this as part of `EnsureCallbacksRegisterd`. Avoids an allocation if a callback isn't registered. Maybe call it `EnsureCallbacksInitialized` or something like that since it's doing more than registration at that point.